test/framework/type_checking.py

```diff
@@ -128,6 +141,10 @@ def test_to_dependency(self):
         self.assertEqual(to_dependency({'lib': '1.2.8'}), {'name': 'lib', 'version': '1.2.8'})
         self.assertEqual(to_dependency({'lib': '1.2.8', 'toolchain': 'GCC, 4.8.2'}), lib_dict)
         self.assertEqual(to_dependency({'lib': '1.2.8', 'toolchain': ['GCC', '4.8.2']}), lib_dict)
+        lib_dict.update({'versionsuffix': ''})
+
+        self.assertEqual(to_dependency(('foo', '1.3')), ('foo','1.3'))
```

Comment:
add a comment here:

```python
# to_dependency doesn't touch values of non-dict type
```

 and add a test case to check that non-dict/tuple/list values are passed untouched (i.e., no error):

```python
self.assertEqual(to_dependency('foo/1.3'), 'foo/1.3')
```